### ISSUE TITLE
Property changes to body's inertia do not update the body's (non-property) inertia

### ISSUE THREAD

**Issue by adamkewley** [opensim-org/opensim-core#3395]
Bug initially filed in `opensim-creator`: https://github.com/ComputationalBiomechanicsLab/opensim-creator/issues/597

tl;dr: @jesse-gilmer noticed that body inertias are not updated after editing the body's `inertia` property in both OpenSim GUI and OpenSim Creator. Reloading the model from disk (i.e. creating an entirely new `OpenSim::Model`, rather than calling `finalizeFromProperties`, or copying, an existing one) does not have this issue.

A quick read through the source code indicates that `OpenSim::Body` relies on dirty-checking `_inertia` at runtime for `NaN` to figure out whether the inertia should be updated from the property. If it is `NaN`, then it updates from the property. If it isn't `NaN`, then it assumes the internal `_inertia` value is up-to-date.

This is a bug, because `_inertia` can be set to not-`NaN` by (e.g.) an earlier call to `getInertia`, but the property can be subsequently edited (e.g. in a GUI) after that point.

The hotfix to this problem is to ensure that `finalizeFromProperties` resets the `_inertia` to `NaN`, such that be whole object behaves "as-if" it was just constructed. A "better" solution would have `extendFinalizeFromProperties` set the `_inertia` member correctly at that time, and the "best" solution would be to not have a `mutable` `SimTK::Inertia` member that is dirty-checked by a `const` getter.

(a quick first PR is likely to be the hotfix, though).
